File: algorithms/2_medium/M/00945_minimum_increment_to_make_array_unique/sort.cpp
class Solution {
public:
    static int minIncrementForUnique(vector<int>& nums) {
        ranges::sort(nums);
        int expect = numeric_limits<int>::min();
        int needs = 0;
        for (const int num : nums) {
            expect = std::max(expect, num);
            needs += expect - num;
            expect += 1;
        }

        return needs;
    }
};

File: algorithms/2_medium/M/00945_minimum_increment_to_make_array_unique/sort.rs
impl Solution {
    pub fn min_increment_for_unique(mut nums: Vec<i32>) -> i32 {
        nums.sort_unstable();
        let mut expect: i32 = i32::MIN;
        let mut needs: i32 = 0;
        for num in nums {
            expect = expect.max(num);
            needs += expect - num;
            expect += 1;
        }

        needs
    }
}
